airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactory.kt

```diff
@@ -396,6 +397,13 @@ class ObjectStoragePathFactory(
             } else {
                 "$prefix/$replacedForPath$replacedForFile"
             }
-        return PathMatcher(Regex(combined), variableToIndex)
+        val withSuffix =
+            if (suffixPattern != null) {
+                variableToIndex["suffix"] = variableToIndex.size + 1
```

Comment:
I am a little bit confused by this. Shouldn't it be `variableToIndex["suffix"] += 1` or something similar?

Reply:
No. This is a table of `variable_name -> regex_matcher_group_index`.

Since we're adding a variable to the end of the list, we give it the next index.